api_app/script_analyzers/classes.py

```diff
@@ -195,15 +198,31 @@ class FileAnalyzer(BaseAnalyzerMixin):
     """
 
     md5: str
-    filepath: str
     filename: str
 
-    def __init__(
-        self, analyzer_name, job_id, fpath, fname, md5, additional_config_params
-    ):
+    @property
+    def filepath(self):
+        if self._filepath is None:
+            self._filepath = self.job_object.file.storage.retrieve(
+                name=self.job_object.file.name, analyzer=self.analyzer_name
+            )
+        return self._filepath
+
+    @property
+    def binary(self):
+        if self._binary is None:
+            self._binary = self.job_object.file.read()
+            self.job_object.file.seek(0)
+        return self._binary
+
+    def __init__(self, analyzer_name, job_id, md5, additional_config_params):
+        from api_app.models import Job
+
+        self.job_object = Job.object_by_job_id(job_id)
         self.md5 = md5
-        self.filepath = fpath
-        self.filename = fname
+        self.filename = self.job_object.file_name
+        self._binary = None
+        self._filepath = None
```

Comment:
We have made similar types of changes in `develop-2` branch. (see [here](https://github.com/intelowlproject/IntelOwl/blob/82a390fde91f9725494efc1e21c829460b1732f6/api_app/analyzers_manager/classes.py#L163-L173)).

I see that you also changed `get_binary` calls in analyzer scripts to use `self.binary` which is great but it might get lost when we merge `develop-2` into `develop`. :crying_cat_face: 

So, IMO, it will be better to have this boto3 storage implementation directly in `develop-2`.

Reply:
If the idea makes sense and it is approved, i can change the target branch and fix the conflicts. DW about that